lib/manager/components/deployment/CustomFileEditor.js

```diff
@@ -40,11 +40,37 @@ export default class CustomFileEditor extends Component<Props, {
 
   _canEdit = () => this.props.customFileEditIdx === null
 
-  _isEditing= () => this.props.idx === this.props.customFileEditIdx
+  /**
+   * Makes sure that the filename is valid. The filename is optional when a uri
+   * is provided, but is required when entering raw input.
+   */
+  _fileNameValid = (): boolean => {
+    const {fileSource, model: customFile} = this.state
+    return (fileSource === 'uri' && !!customFile.uri) ||
+      (!!customFile.contents && !!customFile.filename)
+  }
+
+  /**
+   * Makes sure that a uri or some raw input has been added
+   */
+  _hasContents = (): boolean => {
+    const {model: customFile} = this.state
+    return !!customFile.contents || !!customFile.uri
```

Comment:
Here and above in `_fileNameValid` I would suggest using `Boolean(customFile.contents)` rather than `!!customFile.contents`. I just find it infinitely more readable.

Reply:
I don't think `!!` is unreadable, but whatever, I made the change in 00445ee33e7b02c77a650d4a4098e64b6c528c67